bson/bsoncodec/map_codec.go

```diff
@@ -173,6 +173,7 @@ func (mc *MapCodec) DecodeValue(dc DecodeContext, vr bsonrw.ValueReader, val ref
 	}
 	eTypeDecoder, _ := decoder.(typeDecoder)
 
+	// TODO(GODRIVER-2407): Consider removing this.
```

Comment:
If we end up making another ticket for removing ancestor logic entirely, let's use that ticket number here.